docs/designers-developers/developers/block-api/block-patterns.md

```diff
@@ -0,0 +1,37 @@
+# Patterns (Experimental)
+
+Patterns are predefined block layouts, ready to insert and tweak. 
+
+**Note** Patterns are still under heavy development and the APIs are subject to change.
+
+#### register_pattern
+
+The editor comes with a list of built-in patterns. Theme and plugin authors can register addition custom patterns using the `register_pattern` function.
+
+The `register_pattern` function receives the name of the pattern as the first argument and an array describing properties of the pattern as the second argument.
+
+The properties of the style array must include `name` and `label`: 
+ - `title`: A human-readable title for the pattern.
+ - `content`: Raw HTML content for the pattern.
+
+```php
+register_pattern(
+    'my-plugin/my-awesome-pattern',
+    array(
+		'title'   => __( 'Two buttons', 'my-plugin' ),
+        'content' => "<!-- wp:buttons {\"align\":\"center\"} -->\n<div class=\"wp-block-buttons aligncenter\"><!-- wp:button {\"backgroundColor\":\"very-dark-gray\",\"borderRadius\":0} -->\n<div class=\"wp-block-button\"><a class=\"wp-block-button__link has-background has-very-dark-gray-background-color no-border-radius\">Button One</a></div>\n<!-- /wp:button -->\n\n<!-- wp:button {\"textColor\":\"very-dark-gray\",\"borderRadius\":0,\"className\":\"is-style-outline\"} -->\n<div class=\"wp-block-button is-style-outline\"><a class=\"wp-block-button__link has-text-color has-very-dark-gray-color no-border-radius\">Button Two</a></div>\n<!-- /wp:button --></div>\n<!-- /wp:buttons -->",
```

Comment:
Should we take the opportunity to use the same array syntax as the one we promote in the handbook for CPTs? Side note, it'd be nice to have a formatter that lets you switch between syntaxes quickly.

Reply:
I thought about this, one of the reasons I prefer the HTML is because the "deprecated" and "upgrades" work better. Without the HTML we don't know if a pattern is outdated.